src/Engine/ProtoCore/DSASM/Mirror/ExecutionMirror.cs

```diff
@@ -44,24 +45,91 @@ public ExecutionMirror(ProtoCore.DSASM.Executive exec, ProtoCore.RuntimeCore cor
+        {
+            return GetStringValueImplementation(val,formatSpecifier, heap, langblock, -1, -1, forPrint);
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="val"></param>
+        /// <param name="heap"></param>
+        /// <param name="langblock"></param>
+        /// <param name="forPrint"></param>
+        /// <returns></returns>
+        [Obsolete]
+        //TODO this is used all over the place internally
+        //so it's good to keep it and likely make it internal
+        //until we want to change formatting everywhere we process strings.
         public string GetStringValue(StackValue val, Heap heap, int langblock, bool forPrint = false)
         {
             return GetStringValue(val, heap, langblock, -1, -1, forPrint);
         }
 
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="val"></param>
+        /// <param name="heap"></param>
+        /// <param name="langblock"></param>
+        /// <param name="maxArraySize"></param>
+        /// <param name="maxOutputDepth"></param>
+        /// <param name="forPrint"></param>
+        /// <returns></returns>
+        [Obsolete]
         public string GetStringValue(StackValue val, Heap heap, int langblock, int maxArraySize, int maxOutputDepth, bool forPrint = false)
         {
-            if (formatParams == null)
-                formatParams = new OutputFormatParameters(maxArraySize, maxOutputDepth);
+            //use F6 to match the existing behavior.
+            return GetStringValueImplementation(val, StringUtils.LEGACYFORMATTING, heap, langblock, maxArraySize, maxOutputDepth, forPrint);
+        }
+
+        private string GetStringValueImplementation(StackValue val, string formatSpecifier, Heap heap, int langblock, int maxArraySize,
+           int maxOutputDepth, bool forPrint = false)
+        {
+            var legacyBehavior = formatSpecifier == StringUtils.LEGACYFORMATTING;
+            formatParams ??= new OutputFormatParameters(maxArraySize, maxOutputDepth);
+            //if format is the dynamo format specifier then use prefs format.
+            if (formatSpecifier == StringUtils.DynamoPreferencesNumberFormat)
+            {
+                formatSpecifier = ProtoCore.Mirror.MirrorData.PrecisionFormat;
+            }
 
+           
             if (val.IsInteger)
             {
-                return val.IntegerValue.ToString();
+                //in legacy mode we don't format ints.
+                if (legacyBehavior)
+                {
+                    return val.IntegerValue.ToString();
+                }
+                //TODO I am not sure if it's best to just return null or some message here
+                //or if we should catch the error higher up where we can log a runtime warning.
+                //currently we are doing the latter.
+
+                //return val.IntegerValue.SafeToStringWithFormat(formatSpecifier);
```

Comment:
Can you explain this comment? Is it still relevant?

Reply:
yes, still an open question though I am leaning towards my current solution unless there are objections.
If we encounter an exception formatting, we throw and catch it the entry point of the built-in method, and set warning. I've seen this pattern for other built-ins.